src/lu.jl

```diff
@@ -30,25 +30,54 @@ function Base.show(io::IO, mime::MIME{Symbol("text/plain")}, F::LU)
 end
 
 # LU decomposition
-function lu(A::StaticMatrix, pivot::Union{Val{false},Val{true}}=Val(true))
-    L, U, p = _lu(A, pivot)
+function lu(A::StaticMatrix, pivot::Union{Val{false},Val{true}}=Val(true); check = true)
+    L, U, p = _lu(A, pivot, check)
     LU(L, U, p)
 end
 
 # For the square version, return explicit lower and upper triangular matrices.
 # We would do this for the rectangular case too, but Base doesn't support that.
-function lu(A::StaticMatrix{N,N}, pivot::Union{Val{false},Val{true}}=Val(true)) where {N}
-    L, U, p = _lu(A, pivot)
+function lu(A::StaticMatrix{N,N}, pivot::Union{Val{false},Val{true}}=Val(true);
+            check = true) where {N}
+    L, U, p = _lu(A, pivot, check)
     LU(LowerTriangular(L), UpperTriangular(U), p)
 end
 
-@generated function _lu(A::StaticMatrix{M,N,T}, pivot) where {M,N,T}
+# location of the first zero on the diagonal, 0 when not found
+function __first_zero_on_diagonal(A::StaticMatrix{M,N,T}) where {M,N,T}
```

Comment:
Stylistic nitpick: we've used a single leading underscore elsewhere in this package for internal functions, so I'd somewhat prefer sticking to that.

Reply:
Strictly speaking this is not true (`__lu` is the exception), but of course I am happy to change it.